### DISCUSSION TITLE
Plugin Candidates

### DISCUSSION THREAD

**Discussion by ryanmerolle** [netbox-community/netbox#8364] [General]
As an FYI, if you ever wanted to see feature requests that were tagged as good plugin candidates check this out [plugin candidate](https://github.com/netbox-community/netbox/issues?q=label%3A%22plugin+candidate%22) for just that.

After watching issues, slack google groups, reddit, and etc, I have found some interesting/popular plugin ideas mentioned over the years:

- License Tracking / Management
- Support Contract Tracking / Management
- Telecom Circuit Cost Tracking
- Hardware / Software Lifecycle tracking (EOL, EOS, etc)
- Spare inventory tracking (per site & location leveraging existing devicetype and the upcoming moduletype models)
- RIR REST API support
- Floorplan mapping for locations
- Secret Management integration w/ Hashicorp Vault and/or other products
- Monitoring tool custom link creation & synch (associating monitoring tool device & vm links to NetBox objects where the tool does not use the device name)
- Server Provisioning (tracking shipping details, order number & cabling requirements; identify where to rack based power/space/network ports, and any servers you do not want to be in the same rack as)
- Full cable audits (Could just be a report, but show a single view of where LLDP does not match cables NetBox)

As a reminder if an FR/issue gets suggested for a plugin candidate its often because the idea is either outside of the core functionality of NetBox, thought to be needlessly complex if added to core NetBox, or just not worth the ROI to implement or maintain.  Plugins functions could be pulled into NetBox core if popular and if the plugin approach was well put together.  Example would be social auth inclusion in NetBox in place of need to role you own SSO plugin.

You can find a number of interesting existing NetBox Plugins at the [NetBox Plugin Topic page in GitHub .](https://github.com/topics/netbox-plugin). (Be advised some projects tagged as NetBox Plugins are not, and GitHub does not restrict users from applying said topic to their project.)

**Comment by ZPrimed** [comment]
I'm just one person, but I think some sort of Change Management plugin (for stuff that can't be automated, or things like physical maintenance/etc) could be useful as well. 

I'm looking to try to leverage the Journal Entries to handle it - an Info entry using markdown to link back to a MOP/SOP for the change in Google Docs, and using GDocs "approval" to insure that the MOP is vetted by the appropriate people before it gets executed. Then another Info entry can indicate start of work, and a final Journal Entry can indicate Success or Warning/Critical for a failed MOP.

But I have to imagine it could be done more elegantly with a plugin somehow (then approvals could be tracked in the DB, the MOPs themselves can be DB items, etc).

**Reply by ryanmerolle** [reply]
The model you are describing sounds very similar to what was setup for circuits in https://github.com/jasonyates/netbox-circuitmaintenance

you would need it expanded to more modules.
